src/Microsoft.AspNet.OData.Shared/NavigationPath.cs

```diff
@@ -0,0 +1,98 @@
+﻿// Copyright (c) Microsoft Corporation.  All rights reserved.
+// Licensed under the MIT License.  See License.txt in the project root for license information.
+
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+using System.Collections.ObjectModel;
+using System.Linq;
+using Microsoft.OData.UriParser;
+
+namespace Microsoft.AspNet.OData
+{
+    /// <summary>
+    /// Navigation Path of an OData ID
+    /// </summary>
+    public class NavigationPath
+    {
+        private string navigationPathName;
+        private ReadOnlyCollection<ODataPathSegment> _pathSegments;
+        private ConcurrentDictionary<string, PathItem[]> _pathItemCache = new ConcurrentDictionary<string, PathItem[]>();
+
+        /// <summary>
+        /// Constructor which takes and odataId and creates PathItems
```

Comment:
>which takes and odataId and creates PathItems [](http://example.com/codeflow?start=24&length=45)

please update description

Reply:
Check other class:

mainly use the sentence like: "Initializes a new instance of the <see cref="NavigationPath"/> class."